PRESIDECMS-1808

john5onCheng opened the pull request:
postLoginUrl is not picking up rc scope data, it keep getting from session storage which is not correct

john5onCheng committed:
PRESIDECMS-1808
Ensures rc scope is pickup as priority

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/pixl8/Preside-CMS?pullRequest=747) <br/>All committers have signed the CLA.

DominicWatson commented on system/handlers/Login.cfc:
```diff
@@ -107,7 +107,7 @@ component {
 
 		args.allowRememberMe = _getRememberMeAllowed();
-		args.postLoginUrl    = websiteLoginService.getPostLoginUrl( rc.postLoginUrl ?: event.getCurrentUrl() );
+		args.postLoginUrl    = !isEmpty( rc.postLoginUrl ?: "" ) ? rc.postLoginUrl : websiteLoginService.getPostLoginUrl( rc.postLoginUrl ?: event.getCurrentUrl() );
```
I think this needs a little more work. Looking at the `getPostLoginUrl()` method shows that it sets this value into your session scope if its not already there and this will stop happening with this change. I suggest:

```
args.postLoginUrl = websiteLoginService.getPostLoginUrl(
      explicitUrl = rc.postLoginUrl ?: ""
    , defaultUrl  = event.getCurrentUrl()
);
```

Then change the logic in `websiteLoginService.getPostLoginUrl()` to always use the `explicitUrl` if it is not empty + to set it into session if using it. Otherwise, try using what is in session. Otherwise, use the defaultUrl and set it into the session.

john5onCheng replied:
Hi @DominicWatson  I have made a changes accordingly. It will now pickup explicitURL as priority, save it if available, otherwise look for the value from session or default value

DominicWatson reviewed (changes requested):
Hi Johnson, thanks for this. Please see review comment

john5onCheng committed:
PRESIDECMS-1808
Ensures rc.postLoginURL is saved to session and pick up by login page if value is available, otherwise get value from session
